src/main/scala/cromwell/engine/backend/jes/Run.scala

```diff
@@ -111,19 +83,52 @@ case class Run(jesId: String, pipeline: Pipeline, tag: String) {
     }
   }
 
+  private final def waitForStatus(breakout: RunStatus => Boolean): RunStatus = {
+
+    def checkStatus(previousStatus: Option[RunStatus]): RunStatus = {
+      val currentStatus = status()
+
+      if (!(previousStatus contains currentStatus)) {
+        // If this is the first time checking the status, we log the transition as '-' to 'currentStatus'. Otherwise
+        // just use the state names.
+        val prevStateName = previousStatus map { _.toString } getOrElse "-"
+        Log.info(s"$tag: Status change from $prevStateName to $currentStatus")
+
+        // Update the database state:
+        val newBackendInfo = JesCallBackendInfo(Option(JesId(jesId)), Option(JesStatus(currentStatus.toString)))
+        globalDataAccess.updateExecutionBackendInfo(workflowId, CallKey(call, pipeline.key.index), newBackendInfo)
+      }
+
+      currentStatus
+    }
+
+    val attemptedStatus = TryUtil.retryBlock(
+      fn = checkStatus,
+      isSuccess = breakout,
+      retries = -1,
+      pollingInterval = InitialPollingInterval,
+      pollingBackOffFactor = PollingBackoffFactor,
+      maxPollingInterval = MaximumPollingInterval
+    )
+
+    attemptedStatus match {
+      case util.Success(x) => x
```

Comment:
Because cromwell is so awesome we often end up with mutliple sorts of Success in the same file